js/src/common/Component.ts

```diff
@@ -133,4 +133,31 @@ export default abstract class Component<T extends ComponentAttrs = any> implemen
    * This can be used to assign default values for missing, optional attrs.
    */
   protected static initAttrs<T>(attrs: T): void {}
+
+  // BEGIN DEPRECATED MITHRIL 2 BC LAYER
+
+  /**
+   * Initialize the component's attrs.
+   *
+   * This can be used to assign default values for missing, optional attrs.
+   *
+   * @deprecated, use initAttrs instead.
+   */
+  protected static initProps<T>(attrs: T): void {}
```

Comment:
I am missing code that calls this method when `initAttrs()` does not exist. (Depending on whether and how this can be done with TypeScript, this method stub may then not be needed.)

Reply:
Yeah, just realized this doesn't actually do anything :facepalm:. Fixed!